### DISCUSSION TITLE
TypeError: clientAppEnhance is not a function at createVueApp

### DISCUSSION THREAD

**Discussion by backspacerhino** [vuepress/core#78] [Q&A] [answered]
Hi, I am trying to load my components from docs/.vuepress/components/test.vue

using clientAppEnchange.js file

```
module.exports = {
  clientAppRootComponentFiles: '/docs/.vuepress/components/test.vue',
}
```

but I get this error in the browser

app.js?c6d7:127 Uncaught (in promise) TypeError: clientAppEnhance is not a function
    at createVueApp

Any solution to this? I can't seem to find any other way to load my components

**Comment by Mister-Hope** [comment, marked as answer]
Use `path.resolve`. You should give VuePress the actual path. Unless you project is under root folder (That's obviously not possible), you are giving a wrong path.

Possible config is `path.resolve(__dirname, './components/test.vue)`

**Reply by meteorlxy** [reply]
😅  My bad. I should use `path.resolve()` in docs...

**Reply by backspacerhino** [reply]
I tried 

```
import path from "path"

module.exports = {
  clientAppRootComponentFiles: path.resolve(__dirname, './components/test.vue'),
}
```

but I get this error:
```
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.
```

**Reply by meteorlxy** [reply]
@backspacerhino What the ...?

`clientAppRootComponentFiles` should be placed in `.vuepress/config.js`, not in `clientAppEnhance.js`

**Reply by backspacerhino** [reply]
That makes more sense. I was following [this](https://vuepress2.netlify.app/reference/plugin-api.html#clientapprootcomponentfiles) and it appeared as if it was meant to be put in clientAppEnhance

Now that I did put it in `config.ts`, I am getting 

```
[Vue warn]: Failed to resolve component: Test 
  at <List> 
  at <AsyncComponentWrapper> 
  at <Content> 
  at <Page key="/api/v2/products/products/list.html" > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition key=2 name="fade-slide-y" mode="out-in" > 
  at <Layout> 
  at <AsyncComponentWrapper> 
  at <Vuepress onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< <!----> > > 
  at <RouterView> 
  at <App>
```

list.md


```
# List all products

<Test />
```

Test.vue

```
<template>
	<div>TEST</div>
</template>

<script>
export default {
	name: "Test",
	setup() {
		return {};
	},
};
</script>

<style>

</style>
```

**Reply by Mister-Hope** [reply]
Try to correct fileName casing and try again?
 > Test and test

**Reply by Mister-Hope** [reply]
Wait, are you trying to register it then use it in markdown? You should use clientAppEnhance
to manually register it. 

https://github.com/vuepress-theme-hope/vuepress-theme-hope/blob/v2-dev/packages/components/src/client/appEnhance.ts This is probably something you need.

**Reply by backspacerhino** [reply]
Yes, that's exactly what I need, thank you.
